### PR TITLE
Switch to Smaller-Case in Linkerd2 and Partials Charts

### PR DESCRIPTION
**Pull request by Pothulapati**
First Part of #3822 

This PR
- Updates `values.yaml` and `values-ha.yaml` in `linkerd2` to smaller-case values.
- Updates helm templates in `linkerd-2`, `partials` for the same.
- Update `values.go` to serialize into lower-case.
- Update `patch` sub-chart to use lower-case, along with `inject.go` for the same.

@ihcsim @grampelberg

### CODE REVIEW

File: charts/linkerd2/values.yaml [hunk 1]
```diff
@@ -2,150 +2,150 @@
...
-  Issuer:
-    Scheme: linkerd.io/tls
+identity:
+  issuer:
+    scheme: linkerd.io/tls
 
-    ClockSkewAllowance: 20s
+    clockSkewAllowance: 20s
 
     # must match the expiry date in CrtPEM
```
Comment (alpeb) [inline charts/linkerd2/values.yaml]: `crtPEM`

File: charts/linkerd2/values.yaml [hunk 2]
```diff
@@ -2,150 +2,150 @@
...
-ControllerComponentLabel: linkerd.io/control-plane-component
-ControllerNamespaceLabel: linkerd.io/control-plane-ns
-LinkerdNamespaceLabel: linkerd.io/is-control-plane
+controllerComponentLabel: linkerd.io/control-plane-component
+controllerNamespaceLabel: linkerd.io/control-plane-ns
+linkerdNamespaceLabel: linkerd.io/is-control-plane
 
 # If the namespace is controlled by an external tool or can't be installed with Helm
 # you can disable its installation. In this case:
 # - The namespace created by the external tool must match the Namespace value above
```
Comment (alpeb) [inline charts/linkerd2/values.yaml]: `namespace`

**Comment by alpeb** [comment]
Actually never mind my comment about `TestInstallHelm()`, as you already did that (I was watching a different branch here). I'll run the tests that were skipped here, and if they're green I'll give the :heavy_check_mark:

**Comment by alpeb** [comment]
Turns out the Helm integration test is failing because of this guy in `test_helper.go`:
```
"--set", "Namespace=" + h.namespace,
```
Can you please update it? Thanks!

**Comment by ihcsim** [comment]
This works for me in HA mode too:
```
helm install  --name=linkerd2 --set identity.issuer.crtExpiry="Dec 16 16:14:43 2019" --set-file identity.trustAnchorsPEM=ca.crt --set-file identity.issuer.tls.crtPEM=issuer.crt --set-file identity.issuer.tls.keyPEM=issuer.key --set linkerdVersion=git-0d93fd2a --set proxy.image.version=git-0d93fd2a -f charts/linkerd2/values-ha.yaml charts/linkerd2
```

**Pull request merged**